More than 1 matching staging profile found: Can we choose the longest prefix?

I have the following build error:
```
Caused by: java.lang.IllegalArgumentException: More than 1 matching staging profile found in account ***. Available profiles are: com.squareup, com.squareup.misk

...

Caused by: java.lang.IllegalArgumentException: More than 1 matching staging profile found in account ***. Available profiles are: com.squareup, com.squareup.misk
	at com.vanniktech.maven.publish.nexus.Nexus.findStagingProfile(Nexus.kt:70)
	at com.vanniktech.maven.publish.nexus.Nexus.createRepositoryForGroup(Nexus.kt:95)
	at com.vanniktech.maven.publish.sonatype.SonatypeRepositoryBuildService.createStagingRepository(SonatypeRepositoryBuildService.kt:125)
	at com.vanniktech.maven.publish.sonatype.CreateSonatypeRepositoryTask$CreateStagingRepository.execute(CreateSonatypeRepositoryTask.kt:46)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
```

I think it would be useful if this plugin would attempt to find the _longest prefix_ when searching for staging profiles, in order to resolve ambiguities like the above. We tried to set a group of `com.squareup.misk.root`, and this led to the ambiguity between `com.squareup` and `com.squareup.misk`.

--- a/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
+++ b/nexus/src/main/kotlin/com/vanniktech/maven/publish/nexus/Nexus.kt
@@ -58,26 +58,21 @@ class Nexus(
       return allProfiles[0]
     }
 
-    val candidateProfiles = allProfiles.filter { group == it.name }
-      .ifEmpty { allProfiles.filter { group.startsWith(it.name) } }
-      .ifEmpty { allProfiles.filter { group.commonPrefixWith(it.name).isNotEmpty() } }
-
-    if (candidateProfiles.isEmpty()) {
-      throw IllegalArgumentException(
-        "No matching staging profile found in account $username. It is expected that the account contains a staging " +
-          "profile that matches or is the start of $group. " +
-          "Available profiles are: ${allProfiles.joinToString(separator = ", ") { it.name }}",
-      )
+    val exactMatch = allProfiles.find { group == it.name }
+    if (exactMatch != null) {
+      return exactMatch
     }
 
-    if (candidateProfiles.size > 1) {
-      throw IllegalArgumentException(
-        "More than 1 matching staging profile found in account $username. " +
-          "Available profiles are: ${allProfiles.joinToString(separator = ", ") { it.name }}",
-      )
+    val longestPrefixMatch = allProfiles.maxByOrNull { group.commonPrefixWith(it.name).length }
+    if (longestPrefixMatch != null) {
+      return longestPrefixMatch
     }
 
-    return candidateProfiles[0]
+    throw IllegalArgumentException(
+      "No matching staging profile found in account $username. It is expected that the account contains a staging " +
+        "profile that matches or is the start of $group. " +
+        "Available profiles are: ${allProfiles.joinToString(separator = ", ") { it.name }}",
+    )
   }
 
   private fun createStagingRepository(group: String, profile: StagingProfile): String {

